### PR TITLE
[tests-only]public link share for folder expiration date

### PR DESCRIPTION
**Pull request by swoichha**
### Description
This PR adds tests for sharing folder using public link with expiration date and changing the date.

### Related Issue
https://github.com/owncloud/client/issues/8501

### CODE REVIEW

File: test/gui/tst_sharing/test.feature [hunk 1]
```diff
@@ -80,3 +80,18 @@ Feature: Sharing
...
+        And user "Alice" has set up a client with default settings and password "1234"
+        And user "Alice" has created a public link on the server with following settings
+            | path       | simple-folder |
+            | name       | Public link   |
+            | expireDate | 2038-10-14    |
+            | permissions|read, update, create, delete|
+        When the user opens the public links dialog of "%client_sync_path%/simple-folder" using the client-UI
+        And the user edits the public link named "Public link" of file "simple-folder" changing following
+            | expireDate | 2038-07-21 |
+        Then the fields of the last public link share response of user "Alice" should include on the server
```
Comment (dpakach) [inline test/gui/tst_sharing/test.feature:96]:
```suggestion
        Then the fields of the last public link share response of user "Alice" on the server should include
```
Sounds better this way IMO

File: test/gui/tst_syncing/test.feature [hunk 2]
```diff
@@ -6,7 +6,26 @@ Feature: Syncing files
 
     Scenario: Syncing a file to the server
         Given user "Alice" has been created on the server with default attributes
-        And user "Alice" has set up a client with default settings and polling interval "5000"
+        And user "Alice" has set up a client with these settings and password "1234":
```
Comment (dpakach) [inline test/gui/tst_syncing/test.feature:9]: Why is this change required. Same goes for other changes down below.

Comment (swoichha) [reply test/gui/tst_syncing/test.feature:9]: It must have been changed while I rebased it. I will change it back as it is in master.

**Review by dpakach — COMMENTED** [review]
lgtm 👍🏼

**Review by individual-it — APPROVED** [review]
